### PR TITLE
Improve the way the theme loads translations to not clash with pro

### PR DESCRIPTION
**Pull request by lizconlan**
This is a quick fix to prevent the theme unintentionally removing the
pro translations from the translation repository chain.

It's similar to the [fix for ipvtheme](https://github.com/mysociety/ipvtheme/commit/2e9bc29877c57fa234e03e75c9c8c480a789cb39) but is trying to be mindful of the loading order which affects which set of translations takes precedence.

### CODE REVIEW

**Comment by garethrees** [comment]
> It's similar to the fix for ipvtheme but is trying to be mindful of the loading order which affects which set of translations takes precedence.

Could you describe _how_ the ipvtheme approach affects translation precedence, and why this approach fixes it?

I can see that we've effectively got this…

* IPV: `locale-theme`, `locale`, `locale_alaveteli_pro`
* Belgium: `locale-theme`, `locale_alaveteli_pro`, `locale`

…but without investigating, I don't know what difference that makes.

If the Belgium approach is correct, then do we need to make the same fix to IPV (and indeed Alavetelitheme, so that future pro-enabled installs will get this right)?

**Comment by lizconlan** [comment]
(Happy to update the commit message to reflect this but hashing it out here first)

**tl;dr:** `locale-theme, locale` can be read as `locale-theme` ***overrides*** `locale` so `locale-theme, locale_alaveteli_pro, locale` is subtly different from `locale-theme, locale, locale_alaveteli_pro`. I suspect that that having locale last (annoyingly as just appending locale_alaveteli_pro is easier to write) is the expected outcome.

**Long version:**
The current order (as I understand it) for Belgium is (deliberately ignoring pro for the second) `locale-theme, locale`. This has come up recently when trying to track down an incorrect translation so I can say with reasonable confidence that this is the look up order and that it appears to return as soon as it finds a match (ToDo: what does it do if that first match is empty?). So if the desired phrase is translated in more than one place it will use the first one if finds (the theme version if it's present) and ignores the rest so where we insert the pro translations into that list might be significant (but probably only in a few cases where there are "clashes").

**Comment by garethrees** [comment]
> if the desired phrase is translated in more than one place it will use the first one if finds (the theme version if it's present) 

Yeah, this is correct behaviour.

> so where we insert the pro translations into that list might be significant (but probably only in a few cases where there are "clashes").

Have there been clashes?

In any case, `locale-theme`, `locale_alaveteli_pro`, `locale` _seems_ like the correct ordering to me. @crowbot was there a reason that you did `locale-theme`, `locale`, `locale_alaveteli_pro` in https://github.com/mysociety/ipvtheme/commit/2e9bc29877c57fa234e03e75c9c8c480a789cb39? If not, then I think the approach in this PR looks good.

**Comment by lizconlan** [comment]
> Have there been clashes?

Yes, for Belgium. Clash singular (that we know) but have not got a good feel of whether that's expected or some kind of error. Email thread here (immediately moves on to discuss something else so don't feel obliged to read the lot) https://groups.google.com/a/mysociety.org/d/msg/alaveteli/ymleMrz8tQ4/gaNwXR1SAQAJ

**Comment by lizconlan** [comment]
Merged to the 0.31 branch
